docs/configuration.md

```diff
@@ -0,0 +1,128 @@
+# Connector Configuration Reference
+
+## Configuring Named Connectors
+
+The application can programatically configure named connectors using the java
+connection library.
+
+An application may call `ConnectorRegistry.register()` to register the named
+connector configuration.
+
+```java
+GoogleCredentials myCredentials=GoogleCredentials.create(authToken);
+
+    ConnectorConfig config=new ConnectorConfig.Builder()
+    .withTargetPrincipal("example@project.iam.googleapis.com")
+    .withDelegates(Arrays.asList("delegate@project.iam.googleapis.com"))
+    .withGoogleCredentials(myCredentials)
+    .build();
+
+    ConnectorRegistry.register("my-connector",config);
+```
+
+Then the application would use the connector by adding
+the `cloudSqlNamedConnector`  to the JDBC URL:
+
+```java
+String jdbcUrl="jdbc:mysql:///<DATABASE_NAME>?"+
+    +"cloudSqlInstance=project:region:instance"
+    +"&cloudSqlNamedConnector=my-connector"
+    +"&socketFactory=com.google.cloud.sql.mysql.SocketFactory"
+    +"&user=<DB_USER>&password=<PASSWORD>";
+```
+
+Or using JDBC connection properties, by adding the `cloudSqlNamedConnector`
+property:
+
+```java
+// Set up URL parameters
+Properties connProps=new Properties();
+
+    connProps.setProperty("user","<DB_USER>");
+    connProps.setProperty("password","<PASSWORD>");
+    connProps.setProperty("sslmode","disable");
+    connProps.setProperty("socketFactory","<DRIVER_CLASS>");
+    connProps.setProperty("cloudSqlInstance","project:region:instance");
+
+    connProps.setProperty("cloudSqlNamedConnector","my-connector");
+
+// Initialize connection pool
+    HikariConfig config=new HikariConfig();
+    config.setJdbcUrl("jdbc:mysql:///<DB_NAME>");
+    config.setDataSourceProperties(connProps);
+    config.setConnectionTimeout(10000); // 10s
+
+    HikariDataSource connectionPool=new HikariDataSource(config);
+```
+
+When using a named connector, the JDBC connection will ignore all connector
+configuration properties in the JDBC connection properties, using the named
+connector configuration instead. See the full list
```

Comment:
Should we log an error if people make this mistake? Warning: using a named connector causes the cloudsqlDelegates (or whatever prop) to be ignored

Reply:
I don't think we should log an error. Most people will ignore it.